datalake/backend/tests/test_processing_stack.py

```diff
@@ -0,0 +1,133 @@
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Datalake Compute Environment: %s", datalake_compute_environment)
+
+            return datalake_compute_environment
+
+    raise NoStateMachineFound(APPLICATION_TAG_VAL, ENVIRONMENT_TAG_VAL)
+
+
+def launch_compute_environment_resources(batch_client, datalake_compute_environment):
+    """
+    Trigger immediate launch of computing resources. This will significantly speed up Batch jobs
+    execution.
+    """
+    if (
+        int(datalake_compute_environment["computeResources"]["minvCpus"]) < 1
+        or int(datalake_compute_environment["computeResources"]["desiredvCpus"]) < 1
+    ):
+        batch_client.update_compute_environment(
+            computeEnvironment=datalake_compute_environment["computeEnvironmentName"],
+            computeResources={"minvCpus": 1, "desiredvCpus": 1},
+        )
+
+
+def get_state_machine(stepfunctions_client):
+    state_machines_detection_response = stepfunctions_client.list_state_machines()
+
+    # make sure there are no more records in next pages
+    assert state_machines_detection_response.get("nextToken", None) is None
+
+    for state_machine in state_machines_detection_response["stateMachines"]:
+
+        tags_detection_response = stepfunctions_client.list_tags_for_resource(
+            resourceArn=state_machine["stateMachineArn"]
+        )
+
+        # reformat returned array of tags to dictionary
+        state_machine_tags = {tag["key"]: tag["value"] for tag in tags_detection_response["tags"]}
+
+        if (
+            state_machine_tags[ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and state_machine_tags.get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_state_machine = state_machine
+            logger.info("Datalake State Machine: %s", datalake_state_machine)
+
+            return datalake_state_machine
+
+    raise NoComputeEnvironmentFound(APPLICATION_TAG_VAL, ENVIRONMENT_TAG_VAL)
+
+
+@mark.timeout(1200)
+@mark.infrastructure
+def test_should_successfully_run_dataset_version_creation_process(
+    batch_client, stepfunctions_client
+):
+
+    dataset_path = any_safe_file_path()
+    with S3Object(
+        open("datalake/backend/tests/collection.json", "rb"),
+        f"linz-geospatial-data-lake-{ENV}",  # TODO: change to other bucket
+        f"{dataset_path}/collection.json",
+    ) as s3_object:
+
+        state_machine_input = json.dumps(
+            {
+                "dataset_id": "1xyz",
+                "version_id": "2xyz",
+                "type": "RASTER",
+                "metadata_url": f"s3://{s3_object.bucket_name}/{s3_object.key}",
+            }
+        )
+
+        # speed-up Compute Environment launch
+        datalake_compute_environment = get_compute_environment(batch_client)
+        launch_compute_environment_resources(batch_client, datalake_compute_environment)
```

Comment:
Since this isn't strictly speaking *necessary* for the test it should probably be part of the `stepfunctions_client`.

Reply:
I am not sure how much advantage or troubles it can cause. I would leave it as is for now.